.NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/ChineseDateParserConfiguration.cs

```diff
@@ -1,769 +1,155 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Globalization;
+using System.Collections.Immutable;
 using System.Text.RegularExpressions;
-using Microsoft.Recognizers.Text.Number;
-using Microsoft.Recognizers.Text.Number.Chinese;
-using Microsoft.Recognizers.Text.Utilities;
-using DateObject = System.DateTime;
+using Microsoft.Recognizers.Definitions.Chinese;
 
 namespace Microsoft.Recognizers.Text.DateTime.Chinese
 {
-    public class ChineseDateParserConfiguration : IDateTimeParser
+    public class ChineseDateParserConfiguration : BaseDateTimeOptionsConfiguration, ICJKDateParserConfiguration
     {
         public static readonly string ParserName = Constants.SYS_DATETIME_DATE; // "Date";
 
-        private static readonly int[] MonthMaxDays = { 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
-        private readonly ChineseDateTimeParserConfiguration config;
+        public static readonly List<int> MonthMaxDays = new List<int> { 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
 
-        private readonly IExtractor integerExtractor;
-        private readonly IExtractor ordinalExtractor;
-        private readonly IParser numberParser;
-        private readonly IDateTimeExtractor durationExtractor;
-
-        public ChineseDateParserConfiguration(ChineseDateTimeParserConfiguration configuration)
+        public ChineseDateParserConfiguration(ICJKCommonDateTimeParserConfiguration config)
+             : base(config)
         {
-            config = configuration;
+            IntegerExtractor = config.IntegerExtractor;
+            OrdinalExtractor = config.OrdinalExtractor;
 
-            var numOptions = NumberOptions.None;
-            if ((config.Options & DateTimeOptions.NoProtoCache) != 0)
-            {
-                numOptions = NumberOptions.NoProtoCache;
-            }
+            NumberParser = config.NumberParser;
 
-            var numConfig = new BaseNumberOptionsConfiguration(config.Culture, numOptions);
+            DateExtractor = config.DateExtractor;
+            DurationExtractor = config.DurationExtractor;
 
-            integerExtractor = new IntegerExtractor(numConfig);
-            ordinalExtractor = new OrdinalExtractor(numConfig);
+            DateRegexList = new ChineseDateExtractorConfiguration(this).DateRegexList;

```

Comment:
This seems a bit strange.

Reply:
it is done at the same way in English and the other languages.